demos/python_demos/formula_recognition_demo/formula_recognition_demo.py

```diff
@@ -140,90 +206,98 @@ def build_argparser():
-            output = dec_res[args.output_layer]
-            logit = dec_res[args.logit_layer]
-            logits.append(logit)
-            tgt = np.array([[np.argmax(logit, axis=1)]])
-
-            if tgt[0][0][0] == END_TOKEN:
-                break
-        if args.perf_counts:
-            log.info("Encoder performance statistics")
-            print_stats(exec_net_encoder)
-            log.info("Decoder performance statistics")
-            print_stats(exec_net_decoder)
-
-        logits = np.array(logits)
-        logits = logits.squeeze(axis=1)
-        targets = np.argmax(logits, axis=1)
-        if args.output_file:
-            with open(args.output_file, 'a') as output_file:
-                output_file.write(rec['img_name'] + '\t' +  vocab.construct_phrase(targets) + '\n')
+    args = build_argparser().parse_args()
+    interactive_mode = not (os.path.isdir(args.input) or args.input.endswith('.png') or args.input.endswith('.jpg'))
+    args.__dict__['interactive_mode'] = interactive_mode
+    model = Model(args)
+    if not interactive_mode:
+        non_interactive_demo(model, args)
+        return
+
+    *_, height, width = model.encoder.input_info['imgs'].input_data.shape
+    prev_text = ''
+    demo = InteractiveDemo((height, width), resolution=args.resolution)
+    capture = create_capture(args, demo.resolution)
+    if not capture.isOpened():
+        log.error("Cannot open camera")
+        return()
+    while True:
+        ret, frame = capture.read()
+        bin_crop = demo.get_crop(frame)
+        model_input = prerocess_crop(bin_crop, (height, width), preprocess_type=args.preprocessing_type)
+        frame = demo.put_crop(frame, model_input)
+        model_res = model.infer_async(model_input)
+        if not model_res:
+            phrase = prev_text
         else:
-            print("Image name: {}\nFormula: {}\n".format(rec['img_name'], vocab.construct_phrase(targets)))
+            logits, targets = model_res
+            prob = calculate_probability(logits)
+            log.info("Confidence score is %s", prob)
+            if prob >= args.conf_thresh ** len(logits):
+                log.info("Prediction updated")
+                phrase = model.vocab.construct_phrase(targets)
+            else:
+                log.info("Confidence score is low, prediction is not complete")
+                phrase = ''
+        frame = demo.draw(frame, phrase)
+        prev_text = phrase
+        cv.imshow('Press q to quit.', frame)
+        key = cv.waitKey(1) & 0xFF
+        if key in (ord('Q'), ord('q'), ord('\x1b')):
+            break
+        elif key in (ord('o'), ord('O')):
+            demo.resize_window("decrease")
+        elif key in (ord('p'), ord('P')):
+            demo.resize_window("increase")
 
     log.info("This demo is an API example, for any performance measurements please use the dedicated benchmark_app tool "
              "from the openVINO toolkit\n")
 
 
+def create_capture(args, demo_resolution):
```

Comment:
You should pass just `input` variable, not whole `args`, because you use only `args.input` here.

Reply:
Changed